### PR TITLE
cxxopts: init at 2.2.1

### PR DESCRIPTION
**Pull request by spease**
###### Motivation for this change
cxxopts: init at 2.2.1

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Running on an M1 mac, so I can't virtualize to test under Linux. However, it's a header-only C++ library, and I've enabled tests so hopefully OfBorg will pick up any issues. Packaging was pretty straightforward.

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -8567,6 +8567,12 @@
     githubId = 7669898;
     name = "Katharina Fey";
   };
+  spease = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:8570]: Please put this into a separate commit with the message ``maintainers: add spease``.

File: pkgs/development/libraries/cxxopts/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+stdenv.mkDerivation rec {
+  name = "cxxopts";
+  version = "2.2.1";
+
+  doCheck = true;
+
+  src = fetchFromGitHub {
+    owner = "jarro2783";
+    repo = name;
+    rev = "v" + version;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/cxxopts/default.nix:12]:
```suggestion
    rev = "v${version}";
```

File: pkgs/development/libraries/cxxopts/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{ cmake, fetchFromGitHub, icu, lib, pkg-config, stdenv, enableUnicodeHelp ? false }:
+
+stdenv.mkDerivation rec {
+  name = "cxxopts";
+  version = "2.2.1";
+
+  doCheck = true;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/cxxopts/default.nix:7]:
```suggestion
```

File: pkgs/development/libraries/cxxopts/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    owner = "jarro2783";
+    repo = name;
+    rev = "v" + version;
+    sha256 = "0d3y747lsh1wkalc39nxd088rbypxigm991lk3j91zpn56whrpha";
+  };
+
+  buildInputs = lib.optional enableUnicodeHelp [ icu.dev ];
+  cmakeFlags = lib.optional enableUnicodeHelp [ "-DCXXOPTS_USE_UNICODE_HELP=TRUE" ];
+  nativeBuildInputs = [ cmake ] ++ lib.optional enableUnicodeHelp [ pkg-config ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/cxxopts/default.nix:19]:
```suggestion
  nativeBuildInputs = [ cmake ] ++ lib.optional enableUnicodeHelp [ pkg-config ];

  doCheck = true;

```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110441` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>cxxopts</li>
</ul>
</details>

**Comment by spease** [comment]
I believe I've gotten all the review comments. Let me know if there's anything else.

File: pkgs/development/libraries/cxxopts/default.nix [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  version = "2.2.1";
+
+  src = fetchFromGitHub {
+    owner = "jarro2783";
+    repo = name;
+    rev = "v${version}";
+    sha256 = "0d3y747lsh1wkalc39nxd088rbypxigm991lk3j91zpn56whrpha";
+  };
+
+  buildInputs = lib.optional enableUnicodeHelp [ icu.dev ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/cxxopts/default.nix:14]:
```suggestion
  buildInputs = lib.optional enableUnicodeHelp [ icu ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Can we enable unicode help by default? Please squash the review commit into the package init commit.

File: pkgs/development/libraries/cxxopts/default.nix (hunk 5 above)
Comment (spease) [reply pkgs/development/libraries/cxxopts/default.nix:14]: With this change I get:

```
-- Checking for module 'icu-uc'
--   No package 'icu-uc' found
CMake Error at /nix/store/zqnyzfkyyys15bfgc9sr1z0098fqq6zv-cmake-3.19.2/share/cmake-3.19/Modules/FindPkgConfig.cmake:553 (message):
  A required package was not found
Call Stack (most recent call first):
  /nix/store/zqnyzfkyyys15bfgc9sr1z0098fqq6zv-cmake-3.19.2/share/cmake-3.19/Modules/FindPkgConfig.cmake:741 (_pkg_check_modules_internal)
  CMakeLists.txt:62 (pkg_check_modules)
```

when building enableUnicodeHelp

**Comment by spease** [comment]
Done. Let me know if you need any more changes. This has been a great learning experience for me. I didn't realize GitHub dealt this well with rebasing.

File: pkgs/development/libraries/cxxopts/default.nix (hunk 5 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/cxxopts/default.nix:14]: buildInputs should choose the correct variant. Not sure what wrong here.

Comment (spease) [reply pkgs/development/libraries/cxxopts/default.nix:14]: Is there any documentation on this choosing process?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110441` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>cxxopts</li>
</ul>
</details>

File: pkgs/development/libraries/cxxopts/default.nix (hunk 5 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/cxxopts/default.nix:14]: Couldn't find one.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please change the package init commit to ``cxxopts: init at 2.2.1`` without the v.

**Comment by spease** [comment]
Done
